daemon/cmd/daemon.go

```diff
@@ -985,6 +1006,15 @@ func NewDaemon(ctx context.Context, cancel context.CancelFunc, epMgr *endpointma
 		realIdentityAllocator := d.identityAllocator
 		realIdentityAllocator.InitIdentityAllocator(k8s.CiliumClient(), nil)
 
+		// Preallocate IDs for old CIDRs, must be called after InitIdentityAllocator
+		if option.Config.RestoreState && !option.Config.DryMode && len(d.restoredCIDRs) > 0 {
```

Comment:
nit: Can we simplify this logic, since the other two conditions are implied by the population of `d.restoredCIDRs`?

```suggestion
		// Preallocate IDs for old CIDRs, must be called after InitIdentityAllocator
		if len(d.restoredCIDRs) > 0 {
```

Reply:
Simplified :-)